### DISCUSSION TITLE
Compare pre-release version according to semver

### DISCUSSION THREAD

**Discussion by astellingwerf** [renovatebot/renovate#26258] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.89.6

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

If I use a generic versioning to find the latest version of https://github.com/kptdev/kpt/releases, it will not return kpt v1.0.0-beta.48 but v1.0.0-beta.9, because it is comparing the entire prerelease component as a single string to compare.

Semver spec 2.0.0 article 11.4 describes how the version identifiers should be considered separately until a difference is found. So, in my example, it should:

- compare "beta" to "beta", notice they are equals,
- compare "9" to "48", notice they are different and both are all numeric digits, so compare them numerically, and realize that beta 48 is the newest version.

### Relevant debug logs

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by viceice** [comment]
use semver versioning?

**Reply by astellingwerf** [reply]
In this case, the version string starts with a "v", which I believe is not supported by semver versioning, because it's strict in adhering to the standard and only the standard.

**Reply by rarkins** [reply]
You might find that semver-coerced already works for this use case.

For loose versioning, there's a bug report already for it to be sorted numerically: https://github.com/renovatebot/renovate/issues/26136

But just remember that loose versioning has no concept of "unstable", so it has that downside
